crates/evm/src/decode.rs

```diff
@@ -129,14 +125,17 @@ pub fn decode_revert(
             }
         }
         // keccak(Error(string))
-        [8, 195, 121, 160] => {
-            String::decode(&err[SELECTOR_LEN..]).map_err(|_| eyre::eyre!("Bad string decode"))
-        }
+        [8, 195, 121, 160] => Ok(DynSolType::abi_decode(&DynSolType::String, &err[SELECTOR_LEN..])
+            .map_err(|_| eyre::eyre!("Bad string decode"))
+            .and_then(|v| v.clone().as_str().map(|s| s.to_owned()).wrap_err("Bad string decode"))?
+            .to_owned()),
         // keccak(expectRevert(bytes))
         [242, 141, 206, 179] => {
             let err_data = &err[SELECTOR_LEN..];
             if err_data.len() > 64 {
-                let len = U256::from(&err_data[32..64]).as_usize();
+                let len = U256::try_from_be_slice(&err_data[32..64])
+                    .wrap_err_with(|| "Could not decode slice")?
```

Comment:
we should not be using eyre anywhere in library code, this is a good chacne to get rid of it, also if this is just 2 bytes should just deserialize it as usize in the first place